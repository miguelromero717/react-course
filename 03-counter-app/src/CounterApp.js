import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 0 }) => {

    const [counter, setCounter] = useState(value);

    const increment = () => {
        setCounter(counter + 1);
    }

    const decrement = () => {
        setCounter(counter - 1);
    }

    const reset = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={increment}>+1</button>
            <button onClick={reset}>Reset</button>
            <button onClick={decrement}>-1</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;
