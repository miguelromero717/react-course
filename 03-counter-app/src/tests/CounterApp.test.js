import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';

describe('CounterApp Tests', function () {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<CounterApp/>);
    })

    test('should shows CounterApp correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should shows CounterApp correctly sending a custom value', () => {
        const value = 5;
        const wrapper = shallow(
            <CounterApp value={value}/>
        );

        const valueText = wrapper.find('h2').text();

        expect(Number(valueText)).toBe(value);
        expect(valueText).toBe(String(value));
    });

    test('should increment with button +1', () => {
        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text();

        expect(counterText).toBe('1');
    });

    test('should decrement with button -1', () => {

        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('-1');

    });

    test('should set default value reset button', () => {
        const value = 105;
        const wrapper = shallow(
            <CounterApp value={value}/>
        );

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const textValue = wrapper.find('h2').text();

        expect(textValue).toBe('105');
    });
});