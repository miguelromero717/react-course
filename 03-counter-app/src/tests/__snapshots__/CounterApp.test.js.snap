// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp Tests should shows CounterApp correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
